### ISSUE TITLE
keep getting mongoDB already installed while it is not

### ISSUE THREAD

**Issue by djbobyd** [beclab/Olares#786]
I don't know what happened, maybe it was caused somehow by the image download retries and failures but now I cannot install mongoDB. It is shown as not installed but when I try to install it I get that it is already installed.
I am guessing there is some CRD or some config that is left from the previous attepmts. I just don't know what to delete/change to fix this. 
Can you please point me in the right direction.
Thanks!

**Comment by hysyeah** [comment]
@djbobyd Hello!
Could you please provide more information?Use the following command to get pod log.
`tar -czvf olares.tgz /var/log/pods`

`kubectl get app -o yaml > app.yaml`

`kubectl get appmgr -o yaml > appmgr.yaml`

**Comment by djbobyd** [comment]
Hi,
here are the requested files:
https://cloud.boby.bg/s/cBA4paD56z25nDp 
I couldn't upload them to github directly as they are bigger than the 25 MB limit.

Thanks!

**Comment by djbobyd** [comment]
Hello, I made some progress. I found some old helm install secrets wrongfully showing that mongoDB chart was installed. I deleted them and now I can start the isntallation normally. Unfortunately now I get a cluster in a "CrashLoopBackoff" state. 
the percona middleware throws:

```
2024-12-23T15:04:42.518Z    INFO    [SPEC] reconcile cluster error    {"controller": "psmdb-controller", "object": {"name":"mongo-cluster","namespace":"os-system"}, "namespace": "os-system", "name": "mongo-cluster", "reconcileID": "702d35bc-21fd-466e-9bb0- │
│ 2024-12-23T15:04:42.519Z    ERROR    failed to reconcile cluster    {"controller": "psmdb-controller", "object": {"name":"mongo-cluster","namespace":"os-system"}, "namespace": "os-system", "name": "mongo-cluster", "reconcileID": "702d35bc-21fd-466e-9bb0-8d │
│ github.com/percona/percona-server-mongodb-operator/pkg/controller/perconaservermongodb.(*ReconcilePerconaServerMongoDB).Reconcile                                                                                                                                │
│     /go/src/github.com/percona/percona-server-mongodb-operator/pkg/controller/perconaservermongodb/psmdb_controller.go:491                                                                                                                                       │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile                                                                                                                                                                                   │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:122                                                                                                                                                                 │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler                                                                                                                                                                            │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:323                                                                                                                                                                 │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem                                                                                                                                                                         │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:274                                                                                                                                                                 │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2                                                                                                                                                                               │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:235                                                                                                                                                                 │
│ 2024-12-23T15:05:12.732Z    INFO    [SPEC] reconcile cluster error    {"controller": "psmdb-controller", "object": {"name":"mongo-cluster","namespace":"os-system"}, "namespace": "os-system", "name": "mongo-cluster", "reconcileID": "76f28ee6-8741-49fa-af30- │
│ 2024-12-23T15:05:12.732Z    ERROR    failed to reconcile cluster    {"controller": "psmdb-controller", "object": {"name":"mongo-cluster","namespace":"os-system"}, "namespace": "os-system", "name": "mongo-cluster", "reconcileID": "76f28ee6-8741-49fa-af30-b2 │
│ github.com/percona/percona-server-mongodb-operator/pkg/controller/perconaservermongodb.(*ReconcilePerconaServerMongoDB).Reconcile                                                                                                                                │
│     /go/src/github.com/percona/percona-server-mongodb-operator/pkg/controller/perconaservermongodb/psmdb_controller.go:491                                                                                                                                       │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile                                                                                                                                                                                   │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:122                                                                                                                                                                 │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler                                                                                                                                                                            │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:323                                                                                                                                                                 │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem                                                                                                                                                                         │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:274                                                                                                                                                                 │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2                                                                                                                                                                               │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:235                                                                                                                                                                 │
│ 2024-12-23T15:05:42.756Z    INFO    [SPEC] reconcile cluster error    {"controller": "psmdb-controller", "object": {"name":"mongo-cluster","namespace":"os-system"}, "namespace": "os-system", "name": "mongo-cluster", "reconcileID": "76f28ee6-8741-49fa-af30- │
│ 2024-12-23T15:05:42.756Z    ERROR    failed to reconcile cluster    {"controller": "psmdb-controller", "object": {"name":"mongo-cluster","namespace":"os-system"}, "namespace": "os-system", "name": "mongo-cluster", "reconcileID": "76f28ee6-8741-49fa-af30-b2 │
│ github.com/percona/percona-server-mongodb-operator/pkg/controller/perconaservermongodb.(*ReconcilePerconaServerMongoDB).Reconcile                                                                                                                                │
│     /go/src/github.com/percona/percona-server-mongodb-operator/pkg/controller/perconaservermongodb/psmdb_controller.go:491                                                                                                                                       │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile                                                                                                                                                                                   │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:122                                                                                                                                                                 │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler                                                                                                                                                                            │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:323                                                                                                                                                                 │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem                                                                                                                                                                         │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:274                                                                                                                                                                 │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2                                                                                                                                                                               │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:235
```
the mongo-cluter-cfg-0 logs show:
```
backup-agent 2024/12/23 14:59:12 [entrypoint] pbm-agent-entrypoint cmd: pbm-agent, args: []                                                                                                                                                                      │
│ backup-agent 2024/12/23 14:59:12 [entrypoint] starting `pbm-agent`                                                                                                                                                                                               │
│ mongo-init ++ id -u                                                                                                                                                                                                                                              │
│ mongo-init ++ id -g                                                                                                                                                                                                                                              │
│ mongo-init + install -o 1001 -g 0 -m 0755 -D /ps-entry.sh /opt/percona/ps-entry.sh                                                                                                                                                                               │
│ mongo-init ++ id -u                                                                                                                                                                                                                                              │
│ mongo-init ++ id -g                                                                                                                                                                                                                                              │
│ mongo-init + install -o 1001 -g 0 -m 0755 -D /physical-restore-ps-entry.sh /opt/percona/physical-restore-ps-entry.sh                                                                                                                                             │
│ mongo-init ++ id -u                                                                                                                                                                                                                                              │
│ mongo-init ++ id -g                                                                                                                                                                                                                                              │
│ mongo-init + install -o 1001 -g 0 -m 0755 -D /mongodb-healthcheck /opt/percona/mongodb-healthcheck                                                                                                                                                               │
│ mongo-init ++ id -u                                                                                                                                                                                                                                              │
│ mongo-init ++ id -g                                                                                                                                                                                                                                              │
│ mongo-init + install -o 1001 -g 0 -m 0755 -D /pbm-entry.sh /opt/percona/pbm-entry.sh                                                                                                                                                                             │
│ backup-agent 2024/12/23 14:59:42 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:00:14 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:00:46 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:01:18 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:01:50 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:02:22 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:02:54 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:03:26 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:03:58 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:04:30 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:05:02 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:05:34 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:06:06 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:06:38 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ stream closed EOF for os-system/mongo-cluster-cfg-0 (mongo-init)                                                                                                                                                                                                 │
│ stream closed EOF for os-system/mongo-cluster-cfg-0 (mongod)
```
the mongo-cluster-mongos-0 shows nothing:
```
 mongo-init ++ id -u                                                                                                                                                                                                                                              │
│ mongo-init ++ id -g                                                                                                                                                                                                                                              │
│ mongo-init + install -o 1001 -g 0 -m 0755 -D /ps-entry.sh /opt/percona/ps-entry.sh                                                                                                                                                                               │
│ mongo-init ++ id -u                                                                                                                                                                                                                                              │
│ mongo-init ++ id -g                                                                                                                                                                                                                                              │
│ mongo-init + install -o 1001 -g 0 -m 0755 -D /physical-restore-ps-entry.sh /opt/percona/physical-restore-ps-entry.sh                                                                                                                                             │
│ mongo-init ++ id -u                                                                                                                                                                                                                                              │
│ mongo-init ++ id -g                                                                                                                                                                                                                                              │
│ mongo-init + install -o 1001 -g 0 -m 0755 -D /mongodb-healthcheck /opt/percona/mongodb-healthcheck                                                                                                                                                               │
│ mongo-init ++ id -u                                                                                                                                                                                                                                              │
│ mongo-init ++ id -g                                                                                                                                                                                                                                              │
│ mongo-init + install -o 1001 -g 0 -m 0755 -D /pbm-entry.sh /opt/percona/pbm-entry.sh                                                                                                                                                                             │
│ stream closed EOF for os-system/mongo-cluster-mongos-0 (mongo-init)                                                                                                                                                                                              │
│ stream closed EOF for os-system/mongo-cluster-mongos-0 (mongos)
```
and the mongo-cluster-rs0-0 shows something identical to the config:
```
 mongo-init ++ id -u                                                                                                                                                                                                                                              │
│ mongo-init ++ id -g                                                                                                                                                                                                                                              │
│ mongo-init + install -o 1001 -g 0 -m 0755 -D /ps-entry.sh /opt/percona/ps-entry.sh                                                                                                                                                                               │
│ mongo-init ++ id -u                                                                                                                                                                                                                                              │
│ mongo-init ++ id -g                                                                                                                                                                                                                                              │
│ mongo-init + install -o 1001 -g 0 -m 0755 -D /physical-restore-ps-entry.sh /opt/percona/physical-restore-ps-entry.sh                                                                                                                                             │
│ mongo-init ++ id -u                                                                                                                                                                                                                                              │
│ mongo-init ++ id -g                                                                                                                                                                                                                                              │
│ mongo-init + install -o 1001 -g 0 -m 0755 -D /mongodb-healthcheck /opt/percona/mongodb-healthcheck                                                                                                                                                               │
│ mongo-init ++ id -u                                                                                                                                                                                                                                              │
│ mongo-init ++ id -g                                                                                                                                                                                                                                              │
│ mongo-init + install -o 1001 -g 0 -m 0755 -D /pbm-entry.sh /opt/percona/pbm-entry.sh                                                                                                                                                                             │
│ backup-agent 2024/12/23 14:59:42 [entrypoint] pbm-agent-entrypoint cmd: pbm-agent, args: []                                                                                                                                                                      │
│ backup-agent 2024/12/23 14:59:42 [entrypoint] starting `pbm-agent`                                                                                                                                                                                               │
│ backup-agent 2024/12/23 15:00:12 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:00:44 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:01:16 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:01:48 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:02:20 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:02:52 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:03:24 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:03:56 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:04:28 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:05:00 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:05:32 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:06:04 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:06:36 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:07:08 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:07:40 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:08:12 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ backup-agent 2024/12/23 15:08:44 Retry: create mongo connection: mongo ping: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: localhost:27017, Type: Unknown, Last error: dial tcp [::1]:27017 │
│ stream closed EOF for os-system/mongo-cluster-rs0-0 (mongo-init)                                                                                                                                                                                                 │
│ stream closed EOF for os-system/mongo-cluster-rs0-0 (mongod)
```
but none of these logs are very informative to me...

**Comment by hysyeah** [comment]
Thanks for your information.
I think you can try uninstall mongodb, then delete all files in `/olares/userdata/dbdata/mdbdata` and `/olares/userdata/dbdata/mdbdata-config` if they exist, and then reinstall it.

**Comment by djbobyd** [comment]
Unfortunately this didn't help. Those folders were empty anyway.
I also tried uninstalling the mongodb through helm, deleting all the secrets related to mongodb I could find, cleaning up the batch jobs that were left  and still when I try to install the mongodb server it directly crashes.
I ran out of ideas. I searched for a way to do clean up through the percona operator but couldn't find anything that works.

**Comment by eball** [comment]
Hi @djbobyd , could you please provide the logs of  the container `mongod`, not `backup-agent`, in pods?

```
kubectl logs mongo-cluster-cfg-0 -n os-system -c mongod
kubectl logs mongo-cluster-rs0-0 -n os-system -c mongod
```

And furthermore, did you install the mongodb cluster from the helm chart yourself or from the olares market? If from the helm chart, please also provide your chart to us, we'll do some tests on it.

THX

**Comment by djbobyd** [comment]
Hi, unfortunately this is one of the problems. Those containers do not have any logs. That's why I am clueless as to why they fail. Specifically the mongod containers in the pods fail the second they try to start and do not produce any logs. It might be something missing, some permission issues, or anything else but without the output I cannot see what it is, and that is why I cannot send you any logs. 
All the installs I did through the Olares Market. Never tried manual install. I only uninstalled the chart directly with helm once.

**Comment by eball** [comment]
please try to execute

```
kubectl describe pod mongo-cluster-cfg-0 -n os-system
kubectl describe pod mongo-cluster-rs0-0 -n os-system
```

and post the logs

THX

**Comment by djbobyd** [comment]
Hi @eball @hysyeah ,
I knew it was something stupid. Thanks to the commadns above I noticed that the exit code of the container is 132. When I searched for it I found out that mongo requires AVX. Turns out the generic VM processor used by proxmox doesn't support AVX and this was the reason mongoDB server didn't start. I switched the CPU tpye to host for the Olares VM and now it works just fine :)
Thanks a lot for your time and support!
